src/exporter/DaemonMetricCollector.cc

```diff
@@ -313,28 +285,52 @@ DaemonMetricCollector::get_labels_and_metric_name(std::string daemon_name,
   std::string new_metric_name;
   labels_t labels;
   new_metric_name = metric_name;
-  if (daemon_name.find("rgw") != std::string::npos) {
+  if (daemon_name.find("radosgw") != std::string::npos) {
+    std::size_t pos = daemon_name.find_last_of('.');
+    std::string tmp = daemon_name.substr(pos+1);
+    labels["instance_id"] = quote(tmp);
+  }
+  else if (daemon_name.find("rgw") != std::string::npos) {
     std::string tmp = daemon_name.substr(16, std::string::npos);
     std::string::size_type pos = tmp.find('.');
     labels["instance_id"] = quote("rgw." + tmp.substr(0, pos));
-  } else {
-    labels["ceph_daemon"] = quote(daemon_name);
-    if (daemon_name.find("rbd-mirror") != std::string::npos) {
-      std::regex re(
-          "^rbd_mirror_image_([^/]+)/(?:(?:([^/]+)/"
-          ")?)(.*)\\.(replay(?:_bytes|_latency)?)$");
-      std::smatch match;
-      if (std::regex_search(daemon_name, match, re) == true) {
-        new_metric_name = "ceph_rbd_mirror_image_" + match.str(4);
-        labels["pool"] = quote(match.str(1));
-        labels["namespace"] = quote(match.str(2));
-        labels["image"] = quote(match.str(3));
-      }
+  }
+  else if (daemon_name.find("rbd-mirror") != std::string::npos) {
+    std::regex re(
+        "^rbd_mirror_image_([^/]+)/(?:(?:([^/]+)/"
+        ")?)(.*)\\.(replay(?:_bytes|_latency)?)$");
+    std::smatch match;
+    if (std::regex_search(daemon_name, match, re) == true) {
+      new_metric_name = "ceph_rbd_mirror_image_" + match.str(4);
+      labels["pool"] = quote(match.str(1));
+      labels["namespace"] = quote(match.str(2));
+      labels["image"] = quote(match.str(3));
```

Comment:
Are we also dropping this from the mgr/prometheus? Otherwise, they'll be duplicate entries, won't it?

Additionally, rather than introducing this transformation code in the ceph-exporter, I'd expect this to be provided by the daemons themselves. If not, we're trading a language (Python) which is super-friendly for data processing with bare C++ (tbh, this code is not complicate at all, but since we are introducing labeled perf-counters, why not go all the way?).

Reply:
> Are we also dropping this from the mgr/prometheus? Otherwise, they'll be duplicate entries, won't it?
>

They're already removed. 
> Additionally, rather than introducing this transformation code in the ceph-exporter, I'd expect this to be provided by the daemons themselves. If not, we're trading a language (Python) which is super-friendly for data processing with bare C++ (tbh, this code is not complicate at all, but since we are introducing labeled perf-counters, why not go all the way?).